src/test/java/org/opensearch/jobscheduler/utils/JobDetailsServiceIT.java

```diff
@@ -137,4 +142,57 @@ public void testDeleteNonExistingJobDetails() throws ExecutionException, Interru
         }, exception -> fail(exception.getMessage())));
     }
 
+    public void testUpdateIndexToJobDetails() throws ExecutionException, InterruptedException, TimeoutException {
+        String extensionId = "sample-extension";
+        CompletableFuture<Boolean> inProgressFuture = new CompletableFuture<>();
+        JobDetailsService jobDetailsService = new JobDetailsService(client(), this.clusterService, this.indicesToListen);
+
+        // Index Job Index name, actions, for extensionID
+        jobDetailsService.processJobDetailsForExtensionId(
+            "sample-job-index",
```

Comment:
Suggest making these strings local variables (e.g., `expectedJobIndex`) as they are reused:
 - lines 152-155 and 162
 - lines 169-172
 - should use them at lines 183-186